src/libraries/System.Text.Json/tests/System.Text.Json.SourceGeneration.Tests/Serialization/PropertyVisibilityTests.cs

```diff
@@ -94,15 +94,15 @@ public override async Task HonorCustomConverter_UsingPrivateSetter()
 
             string json = @"{""MyEnum"":""AnotherValue"",""MyInt"":2}";
 
-            // Deserialization baseline, without enum converter, we get JsonException.
+            // Deserialization baseline, without enum converter, we get JsonException. NB order of members in deserialized type is significant for this assertion to succeed.
             await Assert.ThrowsAsync<JsonException>(async () => await JsonSerializerWrapperForString.DeserializeWrapper<StructWithPropertiesWithConverter>(json));
 
-            var obj = await JsonSerializerWrapperForString.DeserializeWrapper<StructWithPropertiesWithConverter>(json, options);
-            Assert.Equal(MySmallEnum.AnotherValue, obj.GetMyEnum);
-            Assert.Equal(0, obj.MyInt); // Private setter can't be used with source-gen.
+            // JsonInclude not supported in source gen.
+            await Assert.ThrowsAsync<InvalidOperationException>(async () => await JsonSerializerWrapperForString.DeserializeWrapper<StructWithPropertiesWithConverter>(json, options));
 
-            // ConverterForInt32 throws this exception.
-            await Assert.ThrowsAsync<NotImplementedException>(async () => await JsonSerializerWrapperForString.SerializeWrapper(obj, options));
+            // JsonInclude on private getters not supported.
```

Comment:
Do we test that there are no exceptions (silent ignore) when `[JsonInclude]` isn't on the property?

Reply:
Yeah I think we do. Here's an example:

https://github.com/dotnet/runtime/blob/0becd27f728f9e1eeb2a168789418ad4191ade99/src/libraries/System.Text.Json/tests/Common/PropertyVisibilityTests.cs#L1994